stdlib/include/llvm/Support/Compiler.h

```diff
@@ -142,6 +142,23 @@
 #define LLVM_ATTRIBUTE_USED
 #endif
 
+/// LLVM_NODISCARD - Warn if a type or return value is discarded.
+
+// Use the 'nodiscard' attribute in C++17 or newer mode.
+#if defined(__cplusplus) && __cplusplus > 201402L && LLVM_HAS_CPP_ATTRIBUTE(nodiscard)
```

Comment:
Since the runtimes are always built with C++ 17, do we need each of these clauses, or can we stick with the `[[nodiscard]]` spelling? This header shouldn't be getting leaked once the runtimes are built.

Reply:
I would rather that we keep it identical to what the branch point has. I don't think that it is a big deal to have the additional spellings - they all do the same thing.

Reply:
Sure, but we will confuse the next person to try to update the forked copy of LLVM as to why this change was made. The syncing of the forked copy is a much more complicated thing than just understanding that the other paths do nothing IMO and so it is a better tradeoff. If you really feel strongly, we could start forking all of the headers and making large changes to "simplify" things but that means that the next time we fork, all that work will need to be rediscovered and potentially break ABI in the process.